velox/expression/tests/SignatureTemplateBinder.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/tests/SignatureTemplateBinder.h"
+
+namespace facebook::velox::test {
+
+namespace {
+
+// Return a random type among those in typeList determined by seed.
+TypePtr randomType(std::mt19937& seed) {
+  static std::vector<TypePtr> typeList{
+      BOOLEAN(), TINYINT(), SMALLINT(), INTEGER(), BIGINT(), REAL(), DOUBLE()};
+  auto index = folly::Random::rand32(typeList.size(), seed);
+  return typeList[index];
+}
+
+std::string fromTypeToBaseName(const TypePtr& type) {
+  std::string typeName = type->kindName();
+  std::transform(
+      typeName.begin(), typeName.end(), typeName.begin(), [](unsigned char c) {
+        return std::tolower(c);
+      });
+  return typeName;
+}
+
+std::optional<TypeKind> fromBaseNameToTypeKind(std::string& typeName) {
+  std::transform(
```

Comment:
Any particular reason not to use boost::algorithm::to_upper_copy like in SignatureBinder?

```
auto typeName = boost::algorithm::to_upper_copy(typeSignature.baseType());
```

Reply:
Thought they were doing the same. Do we prefer boost::algorithm::to_upper_copy() for consistency or some other reason?